test/expected-output-test/expressions/input.ddl

```diff
@@ -0,0 +1,7 @@
+a(k int).
+b(k int, p text, q text, r int).
+c(s text, n int, t text).
+
+Q("test" :: TEXT, 123, id, unnest(y)) * :- a(id), b(id, x,y,z), c(x || y,10,"foo"), [z>100; !z < 20, z < 50].
```

Comment:
Can you add more compound expressions and conditions?  It'd be better to put each in separate rules to make it clear which one fails when it fails.